### PR TITLE
[#77] [KMM] [Android] [CHORE] Set up unit test step for the Android app and shared module

### LINKED ISSUE #77: [Chore] Set up CI/CD workflow for testing the Android app on a PR
# Why
To use GitHub Actions to build and test a PR, the machine must be able to run the KMM application.

# Acceptance Criteria
* A test workflow will be triggered when a PR is created and updated.
* The CI machine must be able to test cases in the `shared` and `androidApp` modules.
* Code coverage will be added to the PR by a bot.

### PR DESCRIPTION
**Pull request by AVI5HEK**
- Close #77 

## What happened 👀

Set up the unit test as well as danger reporting of code analysis and test coverage on a PR.

## Insight 📝

- The detekt Gradle task has been replaced by the Gradle plugin to facilitate reporting.
- Updated the detekt rules to make it work for the Compose project.
- Note 👉🏼: Detekt is failing due to the number of issues exceeding the limit. The issues should be fixed in a separate chore PR since that's not related to the workflow setup.

## Proof Of Work 📹

See the reports commented by Danger below.

### CODE REVIEW

File: .github/workflows/android_and_shared_automatic_code_review.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  pull_request:
+    types: [ opened, reopened, edited, synchronize, ready_for_review ]
+
+jobs:
+  review_pull_request:
+    name: Review pull request
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Cancel Previous Runs
```
Comment (phongvhd93) [inline .github/workflows/android_and_shared_automatic_code_review.yml]: Please use the latest way to cancel the previous build: [ref](https://docs.github.com/en/actions/using-jobs/using-concurrency)

File: Dangerfile
Comment (phongvhd93) [inline Dangerfile:1]: Will it run lint only on the changed files? 🤔

File: shared/shared.podspec [hunk 2]
```diff
@@ -36,4 +36,4 @@ Pod::Spec.new do |spec|
         }
     ]
                 
-end
+end
```
Comment (phongvhd93) [inline shared/shared.podspec]: Please add the blank line here

File: Dangerfile
Comment (AVI5HEK) [reply Dangerfile:1]: No. The whole project will be scanned.

File: Gemfile
Comment (suho) [inline Gemfile:1]: @phongvhd93 As the project has the Gemfile in the root folder, please consider to combine it for both platforms

File: .github/workflows/android_and_shared_automatic_code_review.yml (hunk 1 above)
Comment (AVI5HEK) [reply .github/workflows/android_and_shared_automatic_code_review.yml]: Resolved in a9944c66ab8b8e2fdf33d823c905e87ffe13fa2f

File: shared/shared.podspec (hunk 2 above)
Comment (AVI5HEK) [reply shared/shared.podspec]: Resolved in a9944c66ab8b8e2fdf33d823c905e87ffe13fa2f

File: detekt.yml [hunk 3]
```diff
@@ -64,14 +64,14 @@ complexity:
     active: false
   LargeClass:
     active: true
-    threshold: 400
+    threshold: 150
   LongMethod:
-    active: false
-    threshold: 20
+    active: true
+    threshold: 60
```
Comment (kaungkhantsoe) [inline detekt.yml:70]: Optional: This is gonna be more than `60` 😬. We use `80` in `Gulf` project. We can increase the threshold latter.

**Review by kaungkhantsoe — APPROVED** [review]
Just one concern. The rest LGTM 🚀

File: Gemfile
Comment (phongvhd93) [reply Gemfile:1]: Thanks for reminder, I will update the workflow and configuration for the iOS module

**Review by phongvhd93 — APPROVED** [review]

**Comment by phongvhd93** [comment]
@AVI5HEK, would you check the workflow again? I don't see any green build.

**Review by suho — APPROVED** [review]

**Pull request merged**